lib/index.js

```diff
@@ -190,23 +188,16 @@ class ElectronDownloader {
     })
   }
 
-  createTempDir (cb, onSuccess) {
-    this.tmpdir = path.join(os.tmpdir(), `electron-tmp-download-${process.pid}-${Date.now()}`)
-    fs.mkdirs(this.tmpdir, (err) => {
-      if (err) return cb(err)
-      onSuccess(cb)
-    })
-  }
-
   downloadChecksum (cb) {
-    this.downloadFile(this.checksumUrl, this.checksumFilename, this.cachedChecksum, cb, this.verifyChecksum.bind(this))
+    this.downloadFile(this.checksumUrl, this.cachedChecksum, cb, this.verifyChecksum.bind(this))
   }
 
-  downloadFile (url, filename, cacheFilename, cb, onSuccess) {
-    debug('downloading', url, 'to', this.tmpdir)
+  downloadFile (url, cacheFilename, cb, onSuccess) {
+    const tempFileName = `tmp-${process.pid}-${(tmpFileCounter++).toString(16)}-${path.basename(cacheFilename)}`
```

Comment:
I don't want to use something like [mkdtemp](https://nodejs.org/api/fs.html#fs_fs_mkdtemp_prefix_options_callback) here because only our tool creates temp file here — `process.pid` allows us to be sure that will be no conflicts with other processes. To not complicate and keep it simple.